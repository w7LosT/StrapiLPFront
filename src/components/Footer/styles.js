import styled, { css } from "styled-components";
import {Text} from '../TextComponent/styles';
import {Container as SectionComponent} from '../SectionComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    border-top: 1px solid rgba(0, 0, 0, .3);

    & a{
      color: inherit;
      text-decoration: none;
    }

    & ${Text}{
      font-size: ${theme.fonts.sizes.sm};
    }
  `}
`;
