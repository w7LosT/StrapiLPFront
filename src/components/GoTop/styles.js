import styled, { css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 30px;
    right: 15px;
    width: 50px;
    height: 50px;
    font-size: ${theme.fonts.sizes.lg};
    color: ${theme.default.darkBG};
    background-color: ${theme.default.colorWhite};
    border-radius: 100%;
    border: 2px solid ${theme.default.darkBG};
    box-shadow: 0 3px 6px rgba(0, 0, 0, .4);
    z-index: 6;
    opacity: .7;
    transition: all .3s ease-in-out;

    &:hover{
      opacity: 1;
    }
  `}
`;
