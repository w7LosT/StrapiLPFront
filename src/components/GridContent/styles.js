import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme, background }) => css`
    max-width: 580px;
    margin: 0 auto;
    text-align: center;

    & h1{
      background: none;
      color: ${ background ? theme.default.colorWhite : theme.default.primaryColor };
    }

    & a{
      color: ${ background ? theme.default.colorWhite : theme.default.primaryColor };
    }
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.lg} 0;
  `}
`;
