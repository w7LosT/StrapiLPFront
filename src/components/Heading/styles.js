import styled, { css } from "styled-components";

const titleSize = {
  sm: (theme) => css`
    font-size: ${theme.fonts.sizes.sm};
  `,
  md: (theme) => css`
    font-size: ${theme.fonts.sizes.md};
  `,
  lg: (theme) => css`
    font-size: ${theme.fonts.sizes.lg};
    ${mediaFont(theme)};
  `,
};

const titleCaseFunction = {
  none: (theme) => css`
    text-transform: ${theme.fonts.cases.normal}
  `,
  uppercase: (theme) => css`
    text-transform: ${theme.fonts.cases.uppercase}
  `,
  lowercase: (theme) => css`
    text-transform: ${theme.fonts.cases.lowercase}
  `,
  capitalize: (theme) => css`
    text-transform: ${theme.fonts.cases.capitalize}
  `,
}

const mediaFont = (theme) => css`
  @media ${theme.medias.lteMedium}{
    font-size: ${theme.fonts.sizes.md};
  }
`;

export const Title = styled.h1`
  ${
    ({ theme, colorDark, size, titleCase }) => css `
      color: ${ colorDark ? theme.default.colorWhite : theme.default.mainColor };
      /*background-color: ${colorDark ? theme.default.secondaryBG : theme.default.secondaryBG};*/
      ${titleSize[size](theme)};
      ${titleCaseFunction[titleCase](theme)};
      min-width: 200px;

      > a img{
        margin: 0 auto;
      }

      @media ${theme.medias.lteMedium}{
        margin: 15px 0;
      }
    `
  }
`;
