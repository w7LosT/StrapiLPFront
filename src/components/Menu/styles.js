import styled, { css } from "styled-components";
import { Container as SectionComponent } from '../SectionComponent/styles';

export const isMenuOpen = (theme) => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    display: flex;
    align-items: center;
    justify-content: space-between;
    border-bottom: 2px solid ${theme.default.mainBG};
    background-color: ${theme.default.colorWhite};
    z-index: 5;
    transition: all .3s ease;

    & ${SectionComponent}{
      min-width: 800px;
      padding-top: 12px;
      padding-bottom: 12px;
    }

    @media ${theme.medias.lteMedium}{
      height: 100vh;
      visibility: hidden;
      opacity: 0;
      ${visible && isMenuOpen(theme)};

      & ${SectionComponent}{
        min-width: 100%;
        overflow-y: auto;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;
    align-items: center;
    justify-content: space-between;

    @media ${theme.medias.lteMedium}{
        flex-flow: column wrap;
        justify-content: center;
        height: 100%;
    }
  `}
`;

export const BurgerButton = styled.button`
  ${({ theme, visible }) => css`
    z-index: 6;
    position: fixed;
    top: 0;
    right: 0;
    padding: 5px;
    background-color: ${theme.default.darkBG};
    color: ${theme.default.colorWhite};
    border: none;
    cursor: pointer;
    display: none;
    pointer-events: ${visible ? 'none' : 'all'};

    > svg{
      width: 35px;
      height: 35px;
    }

    @media ${theme.medias.lteMedium}{
      display: flex;
      align-items: center;
      justify-content: center;
    }
  `}
`;
