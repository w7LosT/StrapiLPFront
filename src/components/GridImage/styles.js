import styled, { css } from "styled-components";
import { Title as  HeadingContainer } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    padding: 10px;
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
    padding: ${theme.spacings.xlg} 0;
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    transition: all .3s ease;
    box-shadow: 0 3px 6px rgba(0, 0, 0, .5);

    &:hover{
      transform: scale(1.1);
      box-shadow: 0 6px 8px rgba(0, 0, 0, .5);

    }
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;

  `}
`;
