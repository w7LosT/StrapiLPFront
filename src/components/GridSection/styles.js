import styled, { css } from "styled-components";
import { Title as  HeadingContainer } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 1140px;
    margin: 0 auto;
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
    padding: ${theme.spacings.xlg} 0;
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer}{
      position: relative;
      left: 35px ;
      margin-bottom: 10px;
    }

    ${HeadingContainer}::before{
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xlg};
      top: -1rem;
      left: -3rem;
      transform: rotate(5deg);
    }
  `}
`;
