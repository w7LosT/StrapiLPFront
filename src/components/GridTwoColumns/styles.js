import styled, { css } from "styled-components";
import {Title} from '../Heading/styles';

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    gap: ${theme.spacings.lg};
    align-items: center;
    padding: 25px;

    ${Title}{
      margin-bottom: ${theme.spacings.lg};
      text-align: center;
      background: none !important;
      color: ${background ? theme.default.colorWhite : theme.default.mainColor};
    }

    @media ${theme.medias.lteMedium}{
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css``}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    text-align: center;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 600px;

    @media ${theme.medias.lteMedium}{
      max-width: 300px;
    }
  `}
`;
