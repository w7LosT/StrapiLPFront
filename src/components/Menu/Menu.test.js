import { fireEvent, screen } from "@testing-library/react";
import 'jest-styled-components';
import { renderTheme } from "../../styles/render-theme";
import { Menu } from '.';
import links from '../NavLinks/mock';
import { theme } from "../../styles/theme";

const logoData = {
  text: 'Logo',
  image: 'https://raw.githubusercontent.com/luizomf/strapi_landing_pages/master/public/DADOS/landing-page-2.svg',
  link: 'https://www.google.com',
}

describe('<Menu />', () => {
  it('should render Logo and NavLinks', () => {
    const {container} = renderTheme(<Menu links={links} logoData={logoData} />);
    expect(screen.getByRole('heading', { name: 'Logo' })).toBeInTheDocument();
    expect(screen.getByRole('navigation', { name: 'Main Menu' })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render menu mobile and Button for open and close menu', () => {
    const {container} = renderTheme(<Menu links={links} logoData={logoData} />);
    const button = screen.getByLabelText('Open/Close Menu');
    const menuContainer = button.nextSibling;

    expect(button).toHaveStyle('display: none;');

    expect(button).toHaveStyleRule('display', 'flex', {
      media: theme.medias.lteMedium,
    });

    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.medias.lteMedium,
    });

    expect(screen.getByLabelText('Open Menu')).toBeInTheDocument();

    fireEvent.click(button);

    expect(menuContainer).toHaveStyleRule('opacity', '1', {
      media: theme.medias.lteMedium,
    });

    expect(screen.getByLabelText('Close Menu')).toBeInTheDocument();

    fireEvent.click(menuContainer);

    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.medias.lteMedium,
    });

    expect(screen.getByLabelText('Open Menu')).toBeInTheDocument();
  });

  it('should not render links', () => {
    const {container} = renderTheme(<Menu logoData={logoData} />);

    expect(screen.queryByRole('navigation', { name: 'Main Menu' }).firstChild).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
