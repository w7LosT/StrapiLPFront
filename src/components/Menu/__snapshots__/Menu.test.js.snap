// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should not render links 1`] = `
.c6 {
  color: #0a1128;
  font-size: 16px;
  text-transform: uppercase;
  min-width: 200px;
}

.c6 > a img {
  margin: 0 auto;
}

.c4 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 20px;
}

.c2 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 2px solid #cecece;
  background-color: #fff;
  z-index: 5;
  -webkit-transition: all .3s ease;
  transition: all .3s ease;
}

.c2 .c3 {
  min-width: 800px;
  padding-top: 12px;
  padding-bottom: 12px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 0;
  right: 0;
  padding: 5px;
  background-color: #0a1128;
  color: #fff;
  border: none;
  cursor: pointer;
  display: none;
  pointer-events: all;
}

.c0 > svg {
  width: 35px;
  height: 35px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c7 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 > img {
  max-height: 3rem;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:768px) {
  .c6 {
    margin: 15px 0;
  }
}

@media (max-width:768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }

  .c2 .c3 {
    min-width: 100%;
    overflow-y: auto;
  }
}

@media (max-width:768px) {
  .c5 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    height: 100%;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c8 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
  }
}

<div>
  <button
    aria-label="Open/Close Menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open Menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6"
        >
          <a
            class="c7"
            href="https://www.google.com"
          >
            <img
              alt="Logo"
              src="https://raw.githubusercontent.com/luizomf/strapi_landing_pages/master/public/DADOS/landing-page-2.svg"
            />
          </a>
        </h1>
        <nav
          aria-label="Main Menu"
          class="c8"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Menu /> should render Logo and NavLinks 1`] = `
.c6 {
  color: #0a1128;
  font-size: 16px;
  text-transform: uppercase;
  min-width: 200px;
}

.c6 > a img {
  margin: 0 auto;
}

.c4 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 20px;
}

.c2 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 2px solid #cecece;
  background-color: #fff;
  z-index: 5;
  -webkit-transition: all .3s ease;
  transition: all .3s ease;
}

.c2 .c3 {
  min-width: 800px;
  padding-top: 12px;
  padding-bottom: 12px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 0;
  right: 0;
  padding: 5px;
  background-color: #0a1128;
  color: #fff;
  border: none;
  cursor: pointer;
  display: none;
  pointer-events: all;
}

.c0 > svg {
  width: 35px;
  height: 35px;
}

.c9 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 16px;
  padding: 10px;
  color: #0a1128;
  position: relative;
}

.c9::after {
  content: '';
  position: absolute;
  bottom: 6px;
  left: 0;
  right: 0;
  margin: auto;
  width: 0;
  height: 2.5px;
  background-color: #dc143c;
  -webkit-transition: all .15s linear;
  transition: all .15s linear;
}

.c9:hover::after {
  width: 50%;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c7 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 > img {
  max-height: 3rem;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:768px) {
  .c6 {
    margin: 15px 0;
  }
}

@media (max-width:768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }

  .c2 .c3 {
    min-width: 100%;
    overflow-y: auto;
  }
}

@media (max-width:768px) {
  .c5 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    height: 100%;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c8 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
  }
}

<div>
  <button
    aria-label="Open/Close Menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open Menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6"
        >
          <a
            class="c7"
            href="https://www.google.com"
          >
            <img
              alt="Logo"
              src="https://raw.githubusercontent.com/luizomf/strapi_landing_pages/master/public/DADOS/landing-page-2.svg"
            />
          </a>
        </h1>
        <nav
          aria-label="Main Menu"
          class="c8"
        >
          <a
            class="c9"
            href="#target1"
            role="link"
            target="_self"
          >
            Link 1
          </a>
          <a
            class="c9"
            href="#target2"
            role="link"
            target="_self"
          >
            Link 2
          </a>
          <a
            class="c9"
            href="#target3"
            role="link"
            target="_blank"
          >
            Link 3
          </a>
          <a
            class="c9"
            href="#target4"
            role="link"
            target="_self"
          >
            Link 4
          </a>
          <a
            class="c9"
            href="#target5"
            role="link"
            target="_self"
          >
            Link 5
          </a>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
